pkg/virt-handler/cmd-client/client.go

```diff
@@ -448,14 +446,6 @@ func (c *VirtLauncherClient) SoftRebootVirtualMachine(vmi *v1.VirtualMachineInst
 	return c.genericSendVMICmd("SoftReboot", c.v1client.SoftRebootVirtualMachine, vmi, &cmdv1.VirtualMachineOptions{})
 }
 
-func (c *VirtLauncherClient) ShutdownVirtualMachine(vmi *v1.VirtualMachineInstance) error {
-	return c.genericSendVMICmd("Shutdown", c.v1client.ShutdownVirtualMachine, vmi, &cmdv1.VirtualMachineOptions{})
-}
-
-func (c *VirtLauncherClient) KillVirtualMachine(vmi *v1.VirtualMachineInstance) error {
-	return c.genericSendVMICmd("Kill", c.v1client.KillVirtualMachine, vmi, &cmdv1.VirtualMachineOptions{})
-}
-
```

Comment:
This kind of sucks, but I think we have to keep this around.

Here's the issue. Updates... Think about this scenario.

- Kubevirt v0.48 is deployed
- vmi A is running with virt-launcher v0.48
- KubeVirt is updated to v0.49

at this point, shutting down vmi A properly requires the old virt-handler logic, but that old virt-handler logic won't fire if we strip out the api entirely for 0.49.


Here's a potential solution. Keep the api, but simply make the new virt-launcher no-op the shutdown when the shutdown is already in progress. That way old virt-laucher's still have shutdown called, and the new ones just handle that shutdown command differently.

Reply:
Good point, thanks for raising that. I'm on it.

Perhaps it's reasonable to deprecate it entirely in the future?
How do we keep track of such need-to-be-deprecated code?

Reply:
we don't have a great deprecation policy right now unfortunately :/